### ISSUE TITLE
nixos.org/nix/install doesn't build with libsodium

### ISSUE THREAD

**Issue by grahamc** [NixOS/nix#1449]
A user who downloads the standard Nix installer seems this confusing error:

```
download-from-binary-cache.pl: Nix was not compiled with libsodium, required for signed binary cache support at /nix/store/...
error: substituter ‘download-from-binary-cache.pl’ died unexpectedly
```

This happens if you install Nix and immediately configure it to use signed binary caches. The only two solution are:

1. remove the binary cache signature config option from the Nix config and be happy enough with that, or
2. remove the binary cache signature config option from the Nix config, `nix-env -iA nixpkgs.nix`, and put the config option back in place. This works because the nixpkgs version of nix _does_ have libsodium.

I had expected that the nix distributed by NixOS.org would be functionally the same as the one that comes from nixpkgs, and would have preferred not to have to disable signature checking for a second round of installing nix.

ref: https://github.com/LnL7/nix-darwin/issues/7

**Comment by edolstra** [comment]
I think the only reason for this is that libsodium at some point didn't build on macOS. Now that it does it can be enabled in release.nix in the maintenance branch.

**Comment by LnL7** [comment]
I was wondering why it's disabled by default. The nixos modules set `signed-binary-caches = *` by default so I think it makes sense to do the same on darwin.

**Comment by domenkozar** [comment]
This is now fixed in #1453 ?
